src/gsy_e/gsy_e_core/global_objects_singleton.py

```diff
@@ -26,6 +27,7 @@ class GlobalObjects:
     profiles_handler = ProfilesHandler()
     external_global_stats = ExternalConnectionGlobalStatistics()
     future_market_counter = FutureMarketCounter()
+    scm_external_global_stats = SCMExternalConnectionGlobalStatistics()
```

Comment:
I think that the singleton can be avoided by dependency injection on the simulation class. Essentially the simulation class could potentially keep a member with an instance of this class, and pass it to the classes that need it. The complex part is that external_redis_communicator (that requires this) is part of SimulationConfig, therefore it would also need to be passed in the SimulationConfig class. However I think that it would be much better WRT code grouping than the singleton, and it can be avoided

Reply:
This comment caught me off guard. I remember discussion in length how to make the `grid_stats_tree` available in multiple places of the code and decided for a singleton. Now I spend some time to investigate and indeed it turns out, that we only need the grid_stats_tree only here, as you said: 
`Simulation->SimulationConfig->ExternalConnectionCommunicator->AggregatorHandler`
However, in order to create and update the `grid_stats_tree`, we need to pass the root area like done [here](https://github.com/gridsingularity/gsy-e/blob/36d8a6fe0fe93b69fd5b88ddb44281c1de3d9699/src/gsy_e/gsy_e_core/simulation/simulation.py#L112). 
The class will probably need an `update_area` method then. Let me try this (just fearing that I already tried this in the past which was not successful and lead to the singleton)

Reply:
@spyrostz I made good progress on this, but unfortunately, I found out, that we also need access to the `ExternalConnectionGlobalStatistics` in the external mixing [here](https://github.com/gridsingularity/gsy-e/blob/36d8a6fe0fe93b69fd5b88ddb44281c1de3d9699/src/gsy_e/models/strategy/external_strategies/__init__.py#L472) and [here](https://github.com/gridsingularity/gsy-e/blob/36d8a6fe0fe93b69fd5b88ddb44281c1de3d9699/src/gsy_e/models/strategy/external_strategies/__init__.py#L542) [whereas the second usage seems to be avoidable]
The only solution I see is to create an `EXTERNAL_TICK` event and to controll the dispatching from the simulation class, which I think is an overkill. What do you think.